### DISCUSSION TITLE
Timeline with hours per day chart

### DISCUSSION THREAD

**Discussion by agonzalezm** [d3/d3#3593] [Q&A] [answered]
Is there any timeline per hours like this? can this be done with D3?

![image](https://user-images.githubusercontent.com/2251991/196978879-12161eb2-66e3-462d-a14e-bfce61f5468d.png)

I need to track activity of user per day with a timeline like this that support multiple colours. I used to use google timeline chart but it is discontinued now. Can D3 include a similar chart fucntionality?

**Comment by Fil** [comment, marked as answer]
Yes. You'll probably need to use [d3.scaleTime](https://observablehq.com/@d3/d3-scaletime) to map the time data to positions on the screen.

**Comment by agonzalezm** [comment]
any sample on how to do something similar?
